decoder/DecoderApi_unittest.cpp

```diff
@@ -316,6 +316,70 @@ TEST_P(DecodeApiTest, Flush)
     EXPECT_EQ(outFrames, size);
 }
 
+TEST_P(DecodeApiTest, Flush_NonIDR)
+{
+    NativeDisplay nativeDisplay;
+    memset(&nativeDisplay, 0, sizeof(nativeDisplay));
+    DisplayPtr display = VaapiDisplay::create(nativeDisplay);
+    DecodeSurfaceAllocator* allocator = new DecodeSurfaceAllocator(display);
+
+    SharedPtr<IVideoDecoder> decoder;
+    TestDecodeFrames frames = *GetParam();
+    decoder.reset(createVideoDecoder(frames.getMime()), releaseVideoDecoder);
+    ASSERT_TRUE(bool(decoder));
+
+    decoder->setAllocator(allocator);
+
+    VideoConfigBuffer config;
+    memset(&config, 0, sizeof(config));
+    ASSERT_EQ(YAMI_SUCCESS, decoder->start(&config));
+
+    VideoDecodeBuffer buffer;
+    FrameInfo info;
+    int64_t timeBeforeFlush = 0;
+    int64_t timeCurrent = 0;
+    SharedPtr<VideoFrame> output;
+    int32_t size = frames.getFrameCount();
+    int32_t inFrames = 0;
+    int32_t outFrames = 0;
+
+    for (int i = 0; i < size; i++) {
```

Comment:
This will flush every frame, it does not simulate the real case. It's better to change to this: Suppose we have N frame, we flush the decoder at frame 1 and play to end, seek to begin and flush at frame 2 ..etc